cogeo_mosaic/backend/base.py

```diff
@@ -0,0 +1,31 @@
+"""cogeo_mosaic.backend.base: base Backend class."""
+
+import abc
+import contextlib
+from typing import Dict
+
+
+class BaseBackend(metaclass=contextlib.AbstractContextManager):
+    @abc.abstractmethod
+    def __init__(self, *args, **kwargs):
+        """Load resource"""
```

Comment:
what do you think about adding more properties in the __init__ describing the metadata of the mosaicjson? 

```
def __init__(self, *args, **kwargs):
    self.minzoom: int
    self.maxzoom: int
    self.quadkey_zoom: Optional[int]
    self.bounds: Tuple[float, float, float, float]
    self.center: Optional[Tuple[float, float, int]]
    sefl.version: Optional[str]
    self.name: Optional[str]
    self.mosaicjson_version: str
```

Reply:
I don't really have a preference. Currently they're wrapped in a `dict`, unwrapping them is fine too

Reply:
As it is, essentially the only keys used are `quadkey_zoom`, `minzoom`, and `maxzoom`, right? It might be cleaner to keep them in a `dict`?

Reply:
Yes and no, because it could make it cleaner when we only need to access values 
so instead of doing 
```
return self.mosaic_def["minzoom"] 
```
we do 
```
return self.minzoom
```

Reply:
As I mentioned in another comment:
> Currently the only one of those we actually use is quadkey_zoom (derived from it and minzoom). So I defined only quadkey_zoom in __init__ and if we need more values in the future we can add them then. Sound good?